### PR TITLE
websockets: don't silently discard early data

### PR DESCRIPTION
**Pull request by bmetzdorf**
Signed-off-by: Bjoern Metzdorf <bmetzdorf@apple.com>

*Description*:
This addresses the current websocket behavior where early data (HTTP BODY) is being silently discarded. Now if a websocket client sends an HTTP BODY alongside a websocket upgrade request, then the data gets forwarded to upstream.
Some clients rely on this behavior to save a round trip.

*Risk Level*: Low
Small behavior fix

*Testing*:
Added unit tests and tested manually.

*Docs Changes*:
N/A

*Release Notes*:
still TODO

### CODE REVIEW

**Comment by alyssawilk** [comment]
+1 to integration tests - it'd be good to regression test that this doesn't bypass flow control limits (i.e. we won't queue data infinitely at this layer)

**Comment by bmetzdorf** [comment]
@alyssawilk I added a simple integration test. Let me know what you think.

**Comment by bmetzdorf** [comment]
@alyssawilk Yep, you're absolutely right. Thanks for the clarification :) How does it look now?

File: test/integration/integration_test.cc [hunk 1]
```diff
@@ -286,6 +286,43 @@ TEST_P(IntegrationTest, WebSocketConnectionUpstreamDisconnect) {
...
+  tcp_client = makeTcpConnection(lookupPort("http"));
+  // Send websocket upgrade request
+  tcp_client->write(upgrade_req_str);
+  // Send early data alongside websocket upgrade request
+  tcp_client->write(early_data_req_str);
+  fake_upstream_connection = fake_upstreams_[0]->waitForRawConnection();
+  // The request path gets rewritten from /websocket/test to /websocket.
+  // The size of headers received by the destination is 228 bytes.
+  // Early data is 5 bytes, so the total is 233 bytes.
+  const std::string data = fake_upstream_connection->waitForData(233);
```
Comment (bmetzdorf) [inline test/integration/integration_test.cc]: Should I replace this with

```
const std::string data = fake_upstream_connection->waitForData(228 + early_data_req_str.length());
```
?

**Comment by bmetzdorf** [comment]
@rshriram No worries, you assumption was correct. Not sending a body is the standard behavior.

**Pull request merged**